Controller/CoreController.php

```diff
@@ -158,10 +155,25 @@ protected function getSearchHandler()
      */
     protected function getBaseTemplate()
     {
-        if ($this->getRequest()->isXmlHttpRequest()) {
+        if ($this->getCurrentRequest()->isXmlHttpRequest()) {
             return $this->getAdminPool()->getTemplate('ajax');
         }
 
         return $this->getAdminPool()->getTemplate('layout');
     }
+
+    /**
+     * Get the request object from the container.
+     *
+     * @return Request
+     */
+    private function getCurrentRequest()
+    {
+        // NEXT_MAJOR: simplify this when dropping sf < 2.4
```

Comment:
> What do you mean by "overwriting"?

We create a method named `getRequest` and implement the same logic like in your `getCurrentRequest` without calling any `parent::getRequest`. This won't trigger the original (deprecated) method.

The only problem is the visibility, because `getCurrentRequest` is private and `getRequest` is public.